src/wallet/wallet.cpp

```diff
@@ -2660,6 +2660,12 @@ bool CWallet::CreateTransaction(interfaces::Chain::Lock& locked_chain, const std
 
             // Get the fee rate to use effective values in coin selection
             CFeeRate nFeeRateNeeded = GetMinimumFeeRate(*this, coin_control, &feeCalc);
+            // Do not, ever, assume that it's fine to change the fee rate if the user has explicitly
+            // provided one
+            if (coin_control.m_feerate && nFeeRateNeeded != *coin_control.m_feerate) {
+                strFailReason = "Fee rate needed differs from explicitly provided fee rate";
```

Comment:
"needed" is rather vague. I assume this is based on the mempool relay fee? So maybe: `Fee rate is too low for the current mempool. Increase it to %f sat/b or modify -maxmempool`

Reply:
That's a bit weird, though, for the case when the relay fee is at the minimum. We can't really ask people to change `minrelaytxfee` either, as that would mean their node ends up with non-relayable (to most other nodes) transactions. I don't really see a better alternative though, so I'm going with your version with some tweaks:

I built on this a tiny bit, adding a new optional `FeeEstimateMode` parameter to `CFeeRate::ToString()`, which returns sat/b formatted fee rate if estimate mode is sat/b, otherwise the (pre-existing) btc/kb fee rate. So if a user provides an explicit fee rate using `sat/b`, they get the needed one back in sat/b, otherwise it is expressed in btc/kb.

Reply:
There have been occasional rumblings of allowing a lower relay fee than 1 sat/byte, so it's nice to future proof. It's not an ideal message, but definitely better than silently changing the input.

Reply:
This should cover for the <1sat/b case as well. All the way down to 0.001 sat/b.

Reply:
I updated the above comment a bit to clarify that 1 sat/b is not a hardcoded thing.

Reply:
IIUC coin selection uses `nFeeRateNeeded` here:
```cpp
coin_selection_params.effective_fee = nFeeRateNeeded;
```

This doesn't make sense for transaction with a future nLockTime, and with PSBTs it's debatable, since we don't know when the user intends to broadcast. For `sendmany` and `sendtoaddress` the current behavior is fine, but those don't support either.

For PSBT I'm inclined to think that `effective_fee` should be set to the manual fee. In #16378 I plan to do the same for the new `send` RPC. It returns the hex in that case (as it does with `nLockTime`).

Reply:
So basically, you'd prefer if I could force a transaction using a fee rate that is currently not actually accepted by the mempool? I initially did that, but the result is that the tx goes into the mempool but is rejected by the node's peers, so it is never mined (unless the node itself generates a block). That seems awfully confusing.

Perhaps there should be an exception for the locktime>now case, where a warning is produced but the tx is created since it can't go into the mempool anyway, due to locktime not unfulfilled.

This can be achieved by setting the `fOverrideFeeRate` to `true`, btw.

Reply:
I think the current error is fine when the user intends to broadcast immediately. Only when that's not clearly the case, e.g. for the PSBT methods, we should still allow transaction creation.

Reply:
`fOverrideFeeRate` could do the trick, but it's pretty ugly: it basically "sabotages" `GetMinimumFeeRate`. It seems more readable to skip `GetMinimumFeeRate` altogether when you want to ignore it.